integration-docs: Update Git for new doc format.

Niloth-p opened the pull request:
Fixes: Part of zulip#29592.

Separated out from #32354.

<details>
<summary><h3>Screenshot</h3></summary>

[Doc link](https://zulip.com/integrations/doc/git)
![image](https://github.com/user-attachments/assets/d8bebad7-db34-4c47-9582-8e10497599df)

</details>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

Niloth-p commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,41 @@
 
 1. {!download-python-bindings.md!}
 
-1. {!create-channel.md!}
```
Removed.

Niloth-p commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,41 @@
+    To customize further, refer to the
+    [configuration options](#configuration-options).
 
-    You can specify the branches that will be used for notifications by modifying
```
This was previously in a tip, now has been moved into the configuration options section.

Niloth-p commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,41 @@
-
 1. {!change-zulip-config-file.md!}
 
-    You may also need to change the value of `STREAM_NAME`.
```
Several version of this had been previously discussed.
I had proposed appending this to the above macro in 2 different ways, but neither was perfect, so Lauryn recommended using the sentence version, and Alya recommended improving it further, I have now moved it into the Configuration Options section, since it has a default value set in the script.

Niloth-p commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,41 @@
-    notification.
+1. Symlink the `post-receive` hook of your Git repository by running:
 
-1. Symlink `/usr/local/share/zulip/integrations/git/zulip_git_config.py`
```
This instruction was previously clubbed together as "symlink both".
Alya recommended writing out the instructions, and I was trying to format the instructions for both Windows and Linux/Mac OSes, when I went back to digging into the script and noticed that the configuration file is the same as in other integrations, and the symlink is only for the post-receive git hook.

So, removed that other instruction for symlinking the config file.
And used the Linux command for symlinking the hook to the script, since I noticed that we typically don't mention Windows commands in our integration docs.

Niloth-p commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,41 @@
+
+{end_tabs}
+
+### Configuration options
```
This entire section has only been added in the last update.

Point 1: Inspired by the instruction removed from below the configuration code block, elaborate on it with the default value and a code snippet.

Point 2: Moved as-is from a tip above.

Point 3: Newly added option that I noticed in the config file.

Niloth-p reviewed (commented):
Marked the updates since last review.

laurynmm commented:
@Niloth-p - Thanks for pulling this out and noting all the updates from the previous iterations. I made a few revisions to the configuration options section, as well as the following:
- Removed the link to the configuration options section since the documentation isn't very long or complicated.
- Changed "plugin" to "integration" in the testing tip.

It'd be great if you could review those updates and let me know your thoughts.

<details>
<summary>Updated screenshot</summary>

![Screenshot from 2025-01-06 17-22-53](https://github.com/user-attachments/assets/1bb9ed93-d92e-4e93-b059-414d19599c27)
</details>

Niloth-p commented:
Those changes are great, thanks a lot!
The configuration options are definitely clearer.
I'm sorry for not getting back earlier. I wanted to get some prerequisite PRs merged, which happened just now.


<details>
<summary>Screenshot</summary>

![image](https://github.com/user-attachments/assets/e0088c1b-704f-400d-82f1-732d8b733030)


</details>

### Updates:
- I'm wondering if we need to use two levels of bullet indentation for the Configuration Options, since we haven't been using bullets when there's just one point. I've moved them back, is that okay?
- Used the newly added context properties `integration_path` and `config_file_path`.
- "STREAM_NAME" -> "CHANNEL_NAME" which is accurate only after [#P857](https://github.com/zulip/python-zulip-api/pull/857) is merged.

laurynmm commented:
@Niloth-p - Great! Your updates look good, so once https://github.com/zulip/python-zulip-api/pull/857 is merged/done, then I think you can mark this as ready for integration review at that point.

Niloth-p commented:
<details>
<summary><h3>Screenshot</h3></summary>

![image](https://github.com/user-attachments/assets/19019cc0-f4ff-47b1-aa6e-21ad7ae26ef4)

</details>

### Updates
- Added a missing instruction - Copy the config file to the same directory as the post-receive hook. Hence, re-ordered the 3rd and 4th instructions so that the post-receive hook instruction goes before the config-file instructions.
- Mentioned the trigger "pushes" on the intro line.
- Added a tip linking to the list of version-control integrations.
- Added a warning that this integration is to be run on the server side.
- Added a tip that the post-receive hook gets triggered on every push to the repo. It's already subtly mentioned in the intro line, but this clarifies it a bit more.
- Sample data: Modified the path to the repo's git hooks - `.git/hooks` to `your-repo.git`- because bare-repos do not have a separate `.git` folder inside them.

laurynmm commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,57 @@
+
+!!! warn ""
+
+    This integration is meant to be run on the server-side.
```
I assume this means that the integration needs to be run on the Zulip server? If so, then it would need to be set up by a server administrator, correct? If so, then I think that should be more clearly stated.

laurynmm replied:
I assume that this means it needs to be set up on the Zulip server? And therefore needs to be a server administrator who sets up the integration. If so, that should be more clearly stated. And if not, then this should be clearer on what "server-side" means.

laurynmm commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,57 @@
+
+    Zulip offers integrations for
+    [various repository hosting services][other-integrations], and also
+    supports this direct Git integration for unsupported services.
```
The tip format looks a little odd in the rendered documentation here. I'd maybe just expand the content of the  integration description ...

> Get Zulip notifications for pushes to your Git repositories!
>
> Note that Zulip also has integrations with [GitHub](https://zulip.com/integrations/doc/github), [GitLab](https://zulip.com/integrations/doc/gitlab), and various other [version control hosting services](https://zulip.com/integrations/version-control), which do not require access to the Zulip server to set up.

laurynmm commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,57 @@
+
+* The channel where notifications are sent by updating the value of
+  `CHANNEL_NAME`. By default, notifications are sent to a channel named
+  "commits".
```
Maybe we just leave this as `STREAM_NAME` for now so that we can get the bulk of the documentation updates done. And then this documentation update can be made once https://github.com/zulip/python-zulip-api/pull/857 is merged.

laurynmm reviewed (commented):
@Niloth-p - Thanks for making those updates! I think the tip and warning at the top of the integration don't look great in the rendered documentation. Had a few comments on how we could maybe change that. Let me know if you have any questions!

alya commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,56 @@
+!!! warn ""
+
+    This integration is meant to be executed on your Git server.
+
```
Do we need this warning? Can we just add it to the second instruction?

alya replied:
It seems like that line could say "Zulip server" in the macro?

alya commented on templates/zerver/integrations/git.md:
```diff
@@ -1,31 +1,56 @@
 
-    You may also need to change the value of `STREAM_NAME`.
+        The post-receive hook is triggered on every push to the repository.
 
```
I think this can just be a line of text rather than a tip.

laurynmm commented:
These changes were updated and merged in https://github.com/zulip/python-zulip-api/pull/867, so closing this one.
